src/codecs/avif/mod.rs

```diff
@@ -37,7 +38,20 @@ impl<W: Write> AvifEncoder<W> {
     /// necessary. When data is suitably aligned, i.e. u16 channels to two bytes, then the
     /// conversion may be more efficient.
     pub fn write_image(mut self, data: &[u8], width: u32, height: u32, color: ColorType) -> ImageResult<()> {
-        let config = self.config(color);
+        self.write_image_with_speed_quality(data, width, height, color, 100, 1)
+    }
+
+    /// Encode image data with the indicated color type, quality and speed.
+    ///
+    /// The encoder currently requires all data to be RGBA8, it will be converted internally if
+    /// necessary. When data is suitably aligned, i.e. u16 channels to two bytes, then the
+    /// conversion may be more efficient.
+    pub fn write_image_with_speed_quality(mut self, data: &[u8], width: u32, height: u32, color: ColorType, quality: u8, speed: u8) -> ImageResult<()> {
```

Comment:
Nit on the interface style, I'd prefer storing a `ravif::Config` in the encoder and having another constructor instead. This is already used for options in both `png` and `jpeg`. Another idea would be to provide setters for quality and speed instead of a separate constructor.

Reply:
Definitely agree with you there. The reason I didn't do it was purely down to me being thick and not wanting to interfere with a public struct; it looks like I wasn't paying attention there though as I forgot that in rust properties on a public struct are private unless explicitly declared otherwise. I'll rework that now to use a separate constructor like jpeg - I'm afraid you are going to have quite a bit of commit history added to the repo by the time I'm done.

Reply:
I've now reworked the code to set speed and quality on the constructor instead. Cargo check seems to have broken itself on my machine this morning, but it compiles and works when testing it on a simple application.

The old `config()` function called from `write_image()` was passed the `ColorType`, but made no use of it. I've created an empty function called `set_color()` to replicate this and serve as a reminder, but can delete it if deemed unnecessary.